adr/component_offerings.md

```diff
@@ -0,0 +1,134 @@
+-   Provide quality high level components that allow teams to quickly build applications with the best possible interactions (UX) that are consistent, performant, accessible and extensible without the need of repeating themselves over and over again.
+
+```jsx
+<ClayCard>
+	<ClayCard.Title onClick={handleTitleClick}>
+		<ClayIcon symbol="check" />
+
+		{'Hello World'}
+	</ClayCard.Title>
+
+	<ClayCard.Body onClick={handleBodyClick}>
+		<ClayCard.Description>
+			{'Some cool descriptors'}
+		</ClayCard.Description>
+	</ClayCard.Body>
+</ClayCard>
+
+// vs
+<ClayCard
+	iconTitleLeftSymbol="check"
+	onTitleClick={handleTitleClick}
+	onBodyClick={handleBodyClick}
+	description="foo bar baz"
+	title="hello world"
+/>
+```
+
+## Decision
+
+The decision going forward is to create 2 sets of components
+
+-   low-level composable components such as `<ClayCard.Header>`, `<ClayCard.Footer>`
+-   simple high-level components that cover basic lexicon uses
+
+### What does this look like?
+
+#### Composable:
+
+![Composable Drop Down](./images/component_offerings_2.png)
+
+```jsx
+<ClayDropDown>
+	<ClayDropDown.Search
+		onChange={() => {}}
+		placeholder="Search"
+		spritemap={spritemap}
+		value={''}
+	/>
+
+	<ClayDropDown.ItemList>
+		{items.map((item, i) => (
+			<ClayDropDown.Item href={item.href} key={i}>
+				{item.label}
+			</ClayDropDown.Item>
+		))}
+	</ClayDropDown.ItemList>
+</ClayDropDown>
+```
+
+Composing gives the user more control over the layout and interactions.
+
+#### High-level:
+
+![High Level Drop Down](./images/component_offerings_1.png)
+
+```jsx
+<ClayDropDownWithPagination
+	delta={delta}
+	header="Favorites"
+	items={items} // array of objects with `label` and `href`
+	message={'Showing {0} of {1} Structures'}
+	title="Folder"
+/>
+
+// Compare if this component was composed of low-level components
+<ClayDropDown>
+	<ClayDropDown.Header header="Folder" />
+
+	<ClayDropDown.ItemList>
+		<ClayDropDown.Group header="Favorites">
+			{items.slice(0, page * delta).map((item, i) => (
+				<ClayDropDown.Item href={item.href} key={i}>
+					{item.label}
+				</ClayDropDown.Item>
+			))}
+		</ClayDropDown.Group>
+	</ClayDropDown.ItemList>
+
+	<ClayDropDown.Caption>
+		{`Showing ${page * delta} of ${items.length} structures`}
+	</ClayDropDown.Caption>
+
+	<ClayDropDown.Action onClick={() => setPage(page + 1)}>
+		{'Show More'}
+	</ClayDropDown.Action>
+</ClayDropDown>
+```
+
+You'll notice that the high-level component handles much of the interactions internally. It will handle the click of the `show more`, paginate, and update language as necessary. In fact, even this high-level component will be made from the smaller low-level building blocks.
+
+Finally, both the high-level and low-level components will be exported from each package. The low-level component will be exported as default, but the high-level ones will be exported by name. The high-level components should be put into a
```

Comment:
I'm not sure if I understand correctly.
Are we saying we're going to have two versions of a component, i.e. a high level version and a low level version?
If that's the case it's sounds wrong to me, as we don't want to maintain two versions of the same thing.

Reply:
@julien I think the previous paragraph explains this better but we will have the high-level components following the Lexicon standards with their behaviors and it is building with the low-level ones (For example, Autocomplete has some minimal low level components for deliver markup and its complete logic is made up of DataProvider, LoadingIndicator, Input... see #1995 for example, but this can all be composed and still opens options to customize my own logic with my use cases and we can still get all this composition and deliver wrapped in a high-level component).

Then the cost will be less to keep the two in my view. Since the high-level are composed of low level, we are only offering this option for people who want something fast with the behaviors provided by Lexicon for example.

Reply:
@matuzalemsteles thanks for making that clear. 
That doesn't sound like a bad idea or "duplication" to me.